### PR TITLE
[OB-3663] feat: added uri validation

### PR DESCRIPTION
**Pull request by MAG-mv**
- Update CSP Initialise function to pass in Agent info to prevent ordering issues
- Adding Ping to CSP Initialise to ensure uri is valid
- Added test for failure

### CODE REVIEW

File: Library/src/CSPFoundation.cpp [hunk 1]
```diff
@@ -382,12 +390,19 @@ const csp::common::String& CSPFoundation::GetTenant()
 
 void CSPFoundation::SetClientUserAgentInfo(const csp::ClientUserAgent& ClientUserAgentHeader)
```
Comment (MAG-AdamThorn) [inline Library/src/CSPFoundation.cpp:391]: Do we still need this method if the ClientUserAgent info is being passed in via the Initialize method?

Comment (MAG-mv) [reply Library/src/CSPFoundation.cpp:391]: I kept it just in case clients want to update again. But I can make private if we dont think its necessary

Comment (MAG-mv) [reply Library/src/CSPFoundation.cpp:391]: It might be better to hide it tbf. I dont know why they would want to do this. What do you think?

Comment (MAG-AdamThorn) [reply Library/src/CSPFoundation.cpp:391]: Yes I agree, I think we should make it private

**Pull request closed without merging**